### PR TITLE
FLW-891 Select custom validators screen

### PR DESCRIPTION
**Pull request by stas-litvinov**
A scene containing all the functionality responsible for composing custom validators list:
- List contains all elected validators in current era with their APY, their amount and info button for each
- User can tap on Search button to search/filter by name/address (separate feature)
- User can tap on Filter button to apply filters/sorting (separate feature)
- User can tap on "fill the rest with recommended" to fill the rest available slots with recommended validators
- User can tap on "deselect all" to deselect all
- User can tap on "Show selected: N (max 16)

### CODE REVIEW

File: fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift [hunk 1]
```diff
@@ -1,46 +1,231 @@
...
+
+        var indexes: [Int] = []
+
+        viewModel.cellViewModels =
+            viewModel.cellViewModels.enumerated().map { index, item in
+                var newItem = item
+
+                if newItem.isSelected {
+                    newItem.isSelected = false
+                    indexes.append(index)
```
Comment (ERussel) [inline fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift:92]: functional approach combining with side effects could not be obvious. I suggest first map models and in separate call map selected indexes

File: fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift [hunk 2]
```diff
@@ -1,46 +1,231 @@
...
+        provideViewModels()
+        interactor.setup()
+    }
+
+    // MARK: - Header actions
+
+    func fillWithRecommended() {
+        var index = 0
+
+        while index < recommendedValidatorList.count,
```
Comment (ERussel) [inline fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift:119]: Looks like we could rewrite more concisely in functional way like: ```recommendedValidatorList.filter(...).prefix(maxTargets)```

File: fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListViewController.swift [hunk 3]
```diff
@@ -31,46 +55,226 @@ final class CustomValidatorListViewController: UIViewController, ViewHolder {
...
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection _: Int) -> UIView? {
+        guard let headerViewModel = headerViewModel else { return nil }
+        let headerView: CustomValidatorListHeaderView = tableView.dequeueReusableHeaderFooterView()
+        headerView.bind(viewModel: headerViewModel)
+        return headerView
+    }
+
+    func tableView(_: UITableView, heightForFooterInSection _: Int) -> CGFloat {
+        100.0
```
Comment (ERussel) [inline fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListViewController.swift:264]: Why do we need this dynamic footer? If we need an inset then we could use contentInset property

Comment (stas-litvinov) [reply fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListViewController.swift:264]: You are right. Fixed that on recommended validators screen as well

File: fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift (hunk 1 above)
Comment (stas-litvinov) [reply fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift:92]: Done. Hope, I got the right idea

File: fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift (hunk 2 above)
Comment (stas-litvinov) [reply fearless/Modules/Staking/SelectValidatorsFlow/CustomValidatorList/CustomValidatorListPresenter.swift:119]: Well, let's try

**Review by ERussel — APPROVED** [review]

**Pull request merged**